rclrs/src/parameter.rs

```diff
@@ -3,3 +3,1285 @@ mod value;
+    ///
+    /// Returns Some(T) if a parameter of the requested type exists, None otherwise.
+    pub fn get<T: ParameterVariant>(&self, name: &str) -> Option<T> {
+        let storage = &self.interface._parameter_map.lock().unwrap().storage;
+        let Some(storage) = storage.get(name) else {
+            return None;
+        };
+        match storage {
+            ParameterStorage::Declared(storage) => match &storage.value {
+                DeclaredValue::Mandatory(p) => T::maybe_from(p.read().unwrap().clone()),
+                DeclaredValue::Optional(p) => {
+                    p.read().unwrap().clone().and_then(|p| T::maybe_from(p))
+                }
+                DeclaredValue::ReadOnly(p) => T::maybe_from(p.clone()),
+            },
+            ParameterStorage::Undeclared(value) => T::maybe_from(value.read().unwrap().clone()),
+        }
+    }
+
+    /// Tries to set a parameter with the requested value.
+    ///
+    /// Returns:
+    /// * Ok(()) if setting was successful.
+    /// * Err(DeclarationError::TypeMismatch) if the type of the requested value is different from
+    /// the parameter's type.
+    pub fn set<T: ParameterVariant>(
+        &self,
+        name: &str,
+        value: T,
+    ) -> Result<(), ParameterValueError> {
+        let mut map = self.interface._parameter_map.lock().unwrap();
+        if let Some(entry) = map.storage.get_mut(name) {
+            // If it's declared we can only set if it's the same variant.
+            // Undeclared parameters are dynamic by default
+            match entry {
+                ParameterStorage::Declared(param) => {
+                    if T::kind() == param.kind {
+                        let value = value.into();
+                        param.options.ranges.check_in_range(&value)?;
+                        match &param.value {
+                            DeclaredValue::Mandatory(p) => *p.write().unwrap() = value,
+                            DeclaredValue::Optional(p) => *p.write().unwrap() = Some(value),
+                            DeclaredValue::ReadOnly(_) => {
+                                return Err(ParameterValueError::ReadOnly)
+                            }
+                        }
+                    } else {
+                        return Err(ParameterValueError::TypeMismatch);
+                    }
+                }
+                ParameterStorage::Undeclared(param) => {
+                    *param.write().unwrap() = value.into();
+                }
+            }
+        } else {
+            map.storage.insert(
+                name.to_owned(),
+                ParameterStorage::Undeclared(Arc::new(RwLock::new(value.into()))),
+            );
+        }
+        Ok(())
+    }
+}
+
+pub(crate) struct ParameterInterface {
+    _parameter_map: Arc<Mutex<ParameterMap>>,
+    _override_map: ParameterOverrideMap,
+    allow_undeclared: AtomicBool,
+    //_services: ParameterService,
```

Comment:
Could you remove commented code? I think adding a comment with a `NOTE(luca-della-vedova)` with context would be useful, though.